appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/jwt/JwtTokenParser.java

```diff
@@ -81,8 +84,19 @@ public JwtTokenParser() {
         this(Optional.empty(), Optional.empty());
     }
     
-    public JsonWebTokenImpl parse(String bearerToken, String issuer, PublicKey publicKey) throws Exception {
-        SignedJWT signedJWT = SignedJWT.parse(bearerToken);
+    public void parse(String bearerToken) throws Exception {
+        rawToken = bearerToken;
+        signedJWT = SignedJWT.parse(rawToken);
+        
+        if(!checkIsJWT(signedJWT.getHeader())){
+            throw new IllegalStateException("Not JWT");
+        }
+    }
+    
+    public JsonWebTokenImpl verify(String issuer, PublicKey publicKey) throws Exception {
+        if(signedJWT == null){
```

Comment:
This needs formatting
```suggestion
        if (signedJWT == null) {
```